docs/development/quickstart.rst

```diff
@@ -0,0 +1,365 @@
+create a development environment. If you run into problems at any stage do not hesitate
+to `ask for help <https://www.astropy.org/help.html>`_.
+
+Set up GitHub and Git
+---------------------
+
+To contribute to the astropy repository or build from source, you will need to ensure
+that `Git <https://git-scm.com/>`_ is installed and configured on your machine. `GitHub
+has instructions <https://docs.github.com/en/get-started/quickstart/set-up-git>`__ for
+installing and configuring git, and more details and further resources are available in
+the :ref:`contributing.version_control` section.
+
+Install a C compiler if needed
+------------------------------
+
+How to do this will depend on your platform.
+
+**Windows**
+
+You will need `Build Tools for Visual Studio 2022
+<https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2022>`_.
+
+.. note::
+        You DO NOT need to install Visual Studio 2022.
+        You only need "Build Tools for Visual Studio 2022" found by
+        scrolling down to "All downloads" -> "Tools for Visual Studio".
+        In the installer, select the "Desktop development with C++" Workloads.
+
+Alternative options include:
+
+- Install the necessary components on the command line using `vs_BuildTools.exe <https://learn.microsoft.com/en-us/visualstudio/install/use-command-line-parameters-to-install-visual-studio?source=recommendations&view=vs-2022>`_.
+- Use the `WSL <https://learn.microsoft.com/en-us/windows/wsl/install>`_.
+
+**MacOS**
+
+Install the Developer Tools using ``xcode-select --install``.
+
+Further details and related information can be found at
+https://devguide.python.org/setup/#macos.
+
+**Linux**
+
+For Linux-based installations, you won't have to install any additional components.
+
+.. _contributing.forking:
+
+Create a clone of astropy
+-------------------------
+
+If you have not done so already, you will need your own copy of ``astropy`` to
+build it and/or contribute to the source. Astropy is hosted in the `astropy GitHub repository <https://www.github.com/astropy/astropy>`_ and you need to make a clone.
+
+If you *only want to build astropy* but not contribute, you can run the following and then skip to the :ref:`next section<create-isolated-env>`::
+
+    git clone https://github.com/astropy/astropy.git
+    cd astropy
+
+If you plan to contribute a pull request, you will need a `free GitHub account
+<https://github.com/signup/free>`_.
+
+First, create a `GitHub Fork
+<https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo>`_ by going to the `astropy project page <https://github.com/astropy/astropy>`_
+and hitting the ``Fork`` button.
+
+Next, clone your GitHub fork to your machine:
+
+.. code-block:: shell
+
+    git clone https://github.com/YOUR-USER-NAME/astropy.git
+    cd astropy
+    git remote add upstream https://github.com/astropy/astropy.git
+    git fetch upstream
```

Comment:
`git fetch upstream` is not really necessary for those who just forked.

Reply:
That's what I thought until I tried it... It pulls down all the tags. But I'm fine removing it.

Reply:
Ah, in that case perhaps leave it...